relevanceai/vector_tools/cluster.py

```diff
@@ -436,7 +436,123 @@ def kmeans_cluster(
+        leaf_size: int
+            hdbscan configuration parameter default to 40
+        memory = Memory(cachedir=None)
+            hdbscan configuration parameter on memory management
+        metric: str = "euclidean"
+            hdbscan configuration parameter default to "euclidean"
+        min_samples = None
+            hdbscan configuration parameter default to None
+        p = None
+            hdbscan configuration parameter default to None
+        min_cluster_size:
+            minimum cluster size, 10 by default
+        alias : string
+            "hdbscan", string to be used in naming of the field showing the clustering results
+        cluster_field: string
+            "_cluster_", string to name the main cluster field
+        overwrite : bool
+            False by default, To overwite an existing clusering result
+
+        Example
+        -------------
+
+        >>> client.vector_tools.cluster.hdbscan_cluster(
+            dataset_id="sample_dataset",
+            vector_fields=["sample_1_vector_"] # Only 1 vector field is supported for now
+        )
+        """
+
+        if '.'.join([cluster_field, vector_fields[0], alias]) in self.datasets.schema(dataset_id) and not overwrite:
+            raise ClusteringResultsAlreadyExistsError(EXISTING_CLUSTER_MESSAGE)
+
+        # load the documents
+        docs = self.get_all_documents(dataset_id=dataset_id, filters=filters, select_fields=vector_fields)
+
+        # get vectors
+        if len(vector_fields) == 1:
+            # filtering out entries not containing the specified vector
+            docs = list(filter(DocUtils.list_doc_fields, docs))
+            vectors = self.get_field_across_documents(vector_fields[0], docs)
+        else:
+            raise ValueError("We currently do not support more than 1 vector field yet. This will be supported in the future.")
+
+        # Cluster
+        clusterer = HDBSCAN()
+        clustered_docs = clusterer.fit_transform(
+            vectors= vectors,
+            cluster_args = {
+                "algorithm": algorithm,
+                "alpha": alpha,
+                "approx_min_span_tree": approx_min_span_tree,
+                "gen_min_span_tree": gen_min_span_tree,
+                "leaf_size": leaf_size,
+                "memory": memory,
+                "metric": metric,
+                "min_samples": min_samples,
+                "p": p,
+            },
+            min_cluster_size = min_cluster_size).tolist()
+
+        # Updating the db
+        try:
+            formatted_clustered_docs = [
+                {cluster_field:{vector_fields[0]:{alias:res}},
+                '_id':docs[i]['_id']}
+                for i,res in enumerate(clustered_docs)]
+            results = self.update_documents(dataset_id, formatted_clustered_docs, chunksize = update_documents_chunksize)
+            self.logger.info(results)
+        except Exception as e:
+            self.logger.error(e)
```

Comment:
Why is this in a try except loop?

Reply:
I think it should automatically error when the code is written?

Reply:
It was a mistake to have `self.logger.info(results)` outside try: if there is an exception there is no results to log. So,  I moved it inside to have the results logged when the step is passed successfully.